isamples_metadata/vocabularies/vocabulary_mapper.py

```diff
@@ -29,32 +29,92 @@ def metadata_dict(self) -> dict[str, str]:
 
 
 class ControlledVocabulary:
-    def __init__(self, file_path: str, uri_prefix: str):
-        self.uri_prefix = uri_prefix
-        self.vocabulary_terms_by_key = {}
-        self.vocabulary_terms_by_label = {}
-        with open(file_path, newline="") as csvfile:
-            csvreader = csv.reader(csvfile, delimiter="\t", quoting=csv.QUOTE_NONE)
-            # skip header
-            next(csvreader)
-            for row in csvreader:
-                # The first column is the term key and has a prefix, which we ignore for the purpose of this API,
-                # e.g. mat:anthropogenicmetal -> anthropogenicmetal
-                key = row[0]
-                label = row[1]
-                uri = os.path.join(uri_prefix, key)
-                term = VocabularyTerm(key, label, uri)
-                self.vocabulary_terms_by_key[key] = term
-                self.vocabulary_terms_by_label[label] = term
+    def __init__(self, uijson_dict: dict[str, Any], key_prefix: str):
+        self.vocabulary_terms_by_key: dict[str, VocabularyTerm] = {}
+        self.vocabulary_terms_by_label: dict[str, VocabularyTerm] = {}
+        self._key_prefix = key_prefix
+        self._is_first = True
+        self._process_uijson_dict(uijson_dict)
```

Comment:
Left the previous class in place, but swapped out the guts underneath it to operate on the JSON that we generate from fetching everything from the database.